ethcore/sync/src/chain/handler.rs

```diff
@@ -607,9 +645,10 @@ impl SyncHandler {
 		if sync.snapshot.is_complete() {
 			// wait for snapshot restoration process to complete
 			sync.state = SyncState::SnapshotWaiting;
+		} else {
```

Comment:
Before it would give the task to the "same peer" _and_ to other peers, right. Now it's either or. Curious as to why this change, I don't have enough context to really understand most of this but it just struck me as something I wanted to know more about.

Reply:
So it's only because if you look at the sync_peer method, it returns immediately if the state is Waiting. This is just to prevent this useless check I guess.